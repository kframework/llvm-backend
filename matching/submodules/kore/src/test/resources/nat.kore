[]

module NAT

  sort Nat{} []

  symbol zero{}() : Nat{} []

  symbol succ{}(Nat{}) : Nat{} []

  symbol plus{}(Nat{},Nat{}) : Nat{} []

  axiom{S}
    \exists{S}(
      T:Nat{},
      \equals{Nat{},S}(
        T:Nat{},
        zero{}())) []

  axiom{S}
    \exists{S}(
      T:Nat{},
      \equals{Nat{},S}(
        T:Nat{},
        succ{}(X:Nat{}))) []

  axiom{S}
    \exists{S}(
      T:Nat{},
      \equals{Nat{},S}(
        T:Nat{},
        plus{}(X:Nat{},Y:Nat{}))) []

  axiom{}
    \or{Nat{}}(
      zero{}(),
      \exists{Nat{}}(
        X:Nat{},
        succ{}(X:Nat{}))) []

  axiom{}
    \not{Nat{}}(
      \and{Nat{}}(
        zero{}(),
        \exists{Nat{}}(
          X:Nat{},
          succ{}(X:Nat{})))) []

  axiom{S}
    \equals{Nat{},S}(
      plus{}(X:Nat{},zero{}()),
      X:Nat{}) []

  axiom{S}
    \equals{Nat{},S}(
      plus{}(X:Nat{},succ{}(Y:Nat{})),
      succ{}(plus{}(X:Nat{},Y:Nat{}))) []

endmodule []
