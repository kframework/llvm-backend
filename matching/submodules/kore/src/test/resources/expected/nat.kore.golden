Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "NAT"
            , moduleSentences =
                [ ObjectSentence (SentenceSortSentence SentenceSort
                    { sentenceSortName = (Id "Nat" AstLocationNone) :: Id Object
                    , sentenceSortParameters = []
                    , sentenceSortAttributes = Attributes []
                    })
                , ObjectSentence (SentenceSymbolSentence SentenceSymbol
                    { sentenceSymbolSymbol =
                        Symbol
                            { symbolConstructor = (Id "zero" AstLocationNone) :: Id Object
                            , symbolParams = []
                            }
                    , sentenceSymbolSorts = []
                    , sentenceSymbolReturnSort =
                        SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                    , sentenceSymbolAttributes = Attributes []
                    })
                , ObjectSentence (SentenceSymbolSentence SentenceSymbol
                    { sentenceSymbolSymbol =
                        Symbol
                            { symbolConstructor = (Id "succ" AstLocationNone) :: Id Object
                            , symbolParams = []
                            }
                    , sentenceSymbolSorts =
                        [ SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                        ]
                    , sentenceSymbolReturnSort =
                        SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                    , sentenceSymbolAttributes = Attributes []
                    })
                , ObjectSentence (SentenceSymbolSentence SentenceSymbol
                    { sentenceSymbolSymbol =
                        Symbol
                            { symbolConstructor = (Id "plus" AstLocationNone) :: Id Object
                            , symbolParams = []
                            }
                    , sentenceSymbolSorts =
                        [ SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                        , SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                        ]
                    , sentenceSymbolReturnSort =
                        SortActualSort SortActual
                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                    , sentenceSymbolAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ExistsPattern Exists
                            { existsSort =
                                SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                            , existsVariable =
                                Variable
                                    { variableName = (Id "T" AstLocationNone) :: Id Object
                                    , variableSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    }
                            , existsChild =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (EqualsPattern Equals
                                    { equalsOperandSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    , equalsResultSort =
                                        SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                    , equalsFirst =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                            { variableName = (Id "T" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }))))
                                    , equalsSecond =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "zero" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren = []
                                            }))))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ExistsPattern Exists
                            { existsSort =
                                SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                            , existsVariable =
                                Variable
                                    { variableName = (Id "T" AstLocationNone) :: Id Object
                                    , variableSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    }
                            , existsChild =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (EqualsPattern Equals
                                    { equalsOperandSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    , equalsResultSort =
                                        SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                    , equalsFirst =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                            { variableName = (Id "T" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }))))
                                    , equalsSecond =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "succ" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren =
                                                [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                ]
                                            }))))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ExistsPattern Exists
                            { existsSort =
                                SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                            , existsVariable =
                                Variable
                                    { variableName = (Id "T" AstLocationNone) :: Id Object
                                    , variableSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    }
                            , existsChild =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (EqualsPattern Equals
                                    { equalsOperandSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    , equalsResultSort =
                                        SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                                    , equalsFirst =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                            { variableName = (Id "T" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }))))
                                    , equalsSecond =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "plus" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren =
                                                [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                , Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "Y" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                ]
                                            }))))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters = []
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (OrPattern Or
                            { orSort =
                                SortActualSort SortActual
                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                    , sortActualSorts = []
                                    }
                            , orFirst =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                    { applicationSymbolOrAlias =
                                        SymbolOrAlias
                                            { symbolOrAliasConstructor = (Id "zero" AstLocationNone) :: Id Object
                                            , symbolOrAliasParams = []
                                            }
                                    , applicationChildren = []
                                    }))))
                            , orSecond =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ExistsPattern Exists
                                    { existsSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    , existsVariable =
                                        Variable
                                            { variableName = (Id "X" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }
                                    , existsChild =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "succ" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren =
                                                [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                ]
                                            }))))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters = []
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (NotPattern Not
                            { notSort =
                                SortActualSort SortActual
                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                    , sortActualSorts = []
                                    }
                            , notChild =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (AndPattern And
                                    { andSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    , andFirst =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "zero" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren = []
                                            }))))
                                    , andSecond =
                                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (ExistsPattern Exists
                                            { existsSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            , existsVariable =
                                                Variable
                                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }
                                            , existsChild =
                                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                                    { applicationSymbolOrAlias =
                                                        SymbolOrAlias
                                                            { symbolOrAliasConstructor = (Id "succ" AstLocationNone) :: Id Object
                                                            , symbolOrAliasParams = []
                                                            }
                                                    , applicationChildren =
                                                        [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                            { variableName = (Id "X" AstLocationNone) :: Id Object
                                                            , variableSort =
                                                                SortActualSort SortActual
                                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                                    , sortActualSorts = []
                                                                    }
                                                            }))))
                                                        ]
                                                    }))))
                                            }))))
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (EqualsPattern Equals
                            { equalsOperandSort =
                                SortActualSort SortActual
                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                    , sortActualSorts = []
                                    }
                            , equalsResultSort =
                                SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                            , equalsFirst =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                    { applicationSymbolOrAlias =
                                        SymbolOrAlias
                                            { symbolOrAliasConstructor = (Id "plus" AstLocationNone) :: Id Object
                                            , symbolOrAliasParams = []
                                            }
                                    , applicationChildren =
                                        [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                            { variableName = (Id "X" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }))))
                                        , Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "zero" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren = []
                                            }))))
                                        ]
                                    }))))
                            , equalsSecond =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                    , variableSort =
                                        SortActualSort SortActual
                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                            , sortActualSorts = []
                                            }
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                , MetaSentence (SentenceAxiomSentence SentenceAxiom
                    { sentenceAxiomParameters =
                        [ UnifiedObject (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                        ]
                    , sentenceAxiomPattern =
                        Fix (UnifiedPattern (UnifiedObject (Rotate31 (EqualsPattern Equals
                            { equalsOperandSort =
                                SortActualSort SortActual
                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                    , sortActualSorts = []
                                    }
                            , equalsResultSort =
                                SortVariableSort (SortVariable ((Id "S" AstLocationNone) :: Id Object))
                            , equalsFirst =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                    { applicationSymbolOrAlias =
                                        SymbolOrAlias
                                            { symbolOrAliasConstructor = (Id "plus" AstLocationNone) :: Id Object
                                            , symbolOrAliasParams = []
                                            }
                                    , applicationChildren =
                                        [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                            { variableName = (Id "X" AstLocationNone) :: Id Object
                                            , variableSort =
                                                SortActualSort SortActual
                                                    { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                    , sortActualSorts = []
                                                    }
                                            }))))
                                        , Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "succ" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren =
                                                [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "Y" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                ]
                                            }))))
                                        ]
                                    }))))
                            , equalsSecond =
                                Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                    { applicationSymbolOrAlias =
                                        SymbolOrAlias
                                            { symbolOrAliasConstructor = (Id "succ" AstLocationNone) :: Id Object
                                            , symbolOrAliasParams = []
                                            }
                                    , applicationChildren =
                                        [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (ApplicationPattern Application
                                            { applicationSymbolOrAlias =
                                                SymbolOrAlias
                                                    { symbolOrAliasConstructor = (Id "plus" AstLocationNone) :: Id Object
                                                    , symbolOrAliasParams = []
                                                    }
                                            , applicationChildren =
                                                [ Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "X" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                , Fix (UnifiedPattern (UnifiedObject (Rotate31 (VariablePattern Variable
                                                    { variableName = (Id "Y" AstLocationNone) :: Id Object
                                                    , variableSort =
                                                        SortActualSort SortActual
                                                            { sortActualName = (Id "Nat" AstLocationNone) :: Id Object
                                                            , sortActualSorts = []
                                                            }
                                                    }))))
                                                ]
                                            }))))
                                        ]
                                    }))))
                            }))))
                    , sentenceAxiomAttributes = Attributes []
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }